packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -235,7 +240,8 @@ class CreateCommand extends FlutterCommand {
         );
       }
     }
-    final String projectName = fs.path.basename(projectDirPath);
+
+    final String projectName = argResults['project'] ?? fs.path.basename(projectDirPath);
```

Comment:
I know you didn't move it, but could you move this statement down a few lines to be right before the line that calls _validateProjectName? Better if we keep the creation site close to the first call site that uses it.